scripts/nolints.txt

```diff
@@ -769,13 +769,8 @@ apply_nolint category_theory.monad.comparison_forget doc_blame
 -- category_theory/monad/algebra.lean
 apply_nolint category_theory.monad.algebra has_inhabited_instance
 apply_nolint category_theory.monad.algebra.hom doc_blame has_inhabited_instance
-apply_nolint category_theory.monad.algebra.hom.comp doc_blame
-apply_nolint category_theory.monad.algebra.hom.id doc_blame
-apply_nolint category_theory.monad.forget doc_blame
-apply_nolint category_theory.monad.free doc_blame
-
--- category_theory/monad/basic.lean
-apply_nolint category_theory.monad doc_blame
+apply_nolint category_theory.comonad.coalgebra has_inhabited_instance
```

Comment:
In general we should never add entries to `nolints.txt`. To disable the linter on new declarations, use the `@[nolint XXX]` attribute.

See Rob's comments [here](https://github.com/leanprover-community/mathlib/pull/2080#discussion_r389620262).

Reply:
Ah okay, I was copying what was already there for algebras.

Reply:
Documentation standards used to be (much) lower, but as mathlib grows we've had to crank them up. In order to catch up, it's good if new PRs can improve the doc strings on nearby content as they go.

Reply:
How do I resolve the `has_inhabited_instance` complaint?

Reply:
I think tagging `category_theory.comonad.coalgebra` and `category_theory.comonad.coalgebra.hom` with `@[nolint has_inhabited_instance]` should work.

Reply:
Alternatively, you could actually provided an inhabited instance (which only has to be there for _some_ value of the parameters)... I'm not particularly convinced this linter is a good idea, but I guess it provides at least some assurance a definition is not borked, if you can instantiate it the way you expect to be able to?

Providing `inhabited` instances is often achieved by constructing a trivial example, which is often actually a worthwhile documentation-by-code effort, in any case.

But no one will complain about using `@[nolint has_inhabited_instance]`.